packages/storybook/src/nimble/table-column/menu-button/table-column-menu-button.stories.ts

```diff
@@ -316,34 +322,28 @@ export const menuButtonColumn: StoryObj<MenuButtonColumnTableArgs> = {
         ): void => {
             if (e.detail.newState) {
                 const recordId = e.detail.recordId;
+                const record = storyArgs.currentData.find(
+                    d => d.id === recordId
+                )!;
 
                 const changeFavoriteColor = (color: string): void => {
-                    const newData = storyArgs.currentData.map(d => {
-                        if (d.id !== recordId) {
-                            return d;
-                        }
-                        return {
-                            ...d,
-                            favoriteColor: color
-                        };
-                    });
-                    storyArgs.currentData = newData;
-                    void storyArgs.tableRef.setData(newData);
+                    record.favoriteColor = color;
+                    void storyArgs.tableRef.setData(storyArgs.currentData);
                 };
 
-                const item1 = document.createElement(menuItemTag);
-                item1.textContent = `[${recordId}] Blue`;
-                item1.addEventListener('change', () => changeFavoriteColor('Blue'));
-
-                const item2 = document.createElement(menuItemTag);
-                item2.textContent = `[${recordId}] Green`;
-                item2.addEventListener('change', () => changeFavoriteColor('Green'));
-
-                const item3 = document.createElement(menuItemTag);
-                item3.textContent = `[${recordId}] Purple`;
-                item3.addEventListener('change', () => changeFavoriteColor('Purple'));
-
-                storyArgs.menuRef.replaceChildren(item1, item2, item3);
+                const menuItems: MenuItem[] = [];
```

Comment:
Can't we have this be more template driven? i.e. keep the state on the argsTable (like args.menuCurrentSelectedOption) and have the template use normal template patterns for managing this state?

Reply:
Ah yea, would need a way to mark the properties observable. Think the easy way would be to make a small model class that is assigned to the args table with observable props. Could spend a sec on it but if not straightforward current behavior already an improvement 👍

Reply:
Yessss, it's beautiful, ty :)